src/protocol/server.c

```diff
@@ -56,13 +56,15 @@ protocolServerNew(const String *name, const String *service, IoRead *read, IoWri
     MEM_CONTEXT_NEW_BEGIN("ProtocolServer")
     {
         this = memNew(sizeof(ProtocolServer));
-        this->memContext = memContextCurrent();
 
-        this->name = strDup(name);
-        this->read = read;
-        this->write = write;
-
-        this->handlerList = lstNew(sizeof(ProtocolServerProcessHandler));
+        *this = (ProtocolServer)
+        {
+            .memContext = memContextCurrent(),
```

Comment:
In protocolServerNew, shouldn't we be using MEM_CONTEXT_NEW() instead of memContextCurrent()?